### PR TITLE
Update : customEditor > updateCell after setDat, executeFormula with cell info and manage scroll with column fixed

### PR DESCRIPTION
**Pull request by GBonnaire**
I propose 3 update for Jexcel : 

- customEditor > updateCell after setDat
- executeFormula with cell info 
- manage scroll with column fixed

code in next comments

### CODE REVIEW

File: dist/jexcel.js [hunk 1]
```diff
@@ -0,0 +1,12891 @@
...
+            if (value && value.substr(0, 10) == 'data:image') {
+                var img = document.createElement('img');
+                img.src = value;
+                td.appendChild(img);
+            }
+        } else {
+            if ((''+value).substr(0,1) == '=' && obj.options.parseFormulas == true) {
+                value = obj.executeFormula(value, i, j)
+            }
+            if (obj.options.columns[i].mask) {
```
Comment (GBonnaire) [inline dist/jexcel.js:924]: Add before 

`if (obj.options.columns[i].mask) {`

```
if (obj.options.columns[i].editor && typeof(obj.options.columns[i].editor.updateCell) == 'function' ) {
                value = obj.options.columns[i].editor.updateCell(td, value);
            }
```

File: dist/jexcel.js [hunk 2]
```diff
@@ -0,0 +1,12891 @@
...
+            if (cell && cell.getAttribute) {
+                var x = cell.getAttribute('data-x');
+                var y = cell.getAttribute('data-y');
+            }
+
+            // Update cell
+            if (x != null && y != null) {
+                records.push(obj.updateCell(x, y, value, force));
+
+
```
Comment (GBonnaire) [inline dist/jexcel.js:4320]: In obj.executeFormula = function(expression, x, y) {

Replace 
`var evalstring = '';`
by : 
`var evalstring = 'var cell = {col:'+(x+1)+',row:'+(y+1)+', name:"'+jexcel.getColumnNameFromId([x, y])+'"};';`


In use in custom formula on cell.row, cell.col, cell.name

Comment (GBonnaire) [inline dist/jexcel.js:2516]: Add for column fixed with plugin tableHeadFixer.js
[tableHeadFixer.js.txt](https://github.com/paulhodel/jexcel/files/3767250/tableHeadFixer.js.txt)


add before in  :  obj.updateScroll = function(direction) {
```
// Top position check
if (y > (obj.content.scrollTop + 30) && y < (obj.content.scrollTop + h1)) {
```
```
var correction = 0;
        for(var ite_col in obj.records[0]) {
            if(obj.records[0][ite_col].style.position=="relative") {
                correction = Math.max(correction,obj.records[0][ite_col].getBoundingClientRect().left+obj.records[0][ite_col].getBoundingClientRect().width);
            } else {
                break;
            }
        }
        if(x2<correction) {
            x = x-correction-obj.content.scrollLeft+parseInt(obj.records[0][ite_col].style.left);
        }
```

**Comment by pphod** [comment]
There are conflicts, 42 files changed... and your PR is based on an older version. Basically you are bringing files already achieved. Master should contain only v3 now. Ideally, firstly get the most recent version from the master, then apply the changes and improvements only and PR again.

**Comment by pphod** [comment]
I will take a look on your messages to see the best approach to include that in the most recent version.

**Pull request closed without merging**